cider/exec/operator/aggregate/CiderAggHashTable.cpp

```diff
@@ -170,11 +170,8 @@ uint64_t CiderHasher::updateHashMode(const uint64_t entry_num_limit,
   return entry_num_limit_max;
 }
 
-void CiderHasher::copyStringToLocal(UStringVal* uString) {
+void CiderStringHasher::copyStringToLocal(UStringVal* uString) {
   auto size = uString->size();
-  if (size <= sizeof(int64_t)) {
```

Comment:
I think we should keep it since we don't need to copy a string whose length is less than the size of a pointer to `uStringValStorage_`.

Reply:
Based on my understanding, if keep this, it will not save a copy in `uStringValStorage_`,  and cannot find the data from `uStringValStorage_` when use(look up) this string via id.

Reply:
When we call `lookupValueById`, it will always find string through `stringCacheVec` no matter the string length is above or below 8 bytes. 

Besides, `uStringValStorage_` is designed to store long strings in a contiguous block of memory.
Ⅰ. When the length of string is less than or equal to 8 bytes (size of a pointer), we store the pointer to `stringCacheVec` directly. The memory to this pointer is allocated in `unordered_flat_set` when doing `insert`.
Ⅱ. When the length of string is more than 8 bytes, we store them into one or more contiguous blocks of memory. New memory will be allocated and pointers will be changed in `copyStringToLocal`.

Reply:
For less than or equal 8 bytes scenario, the pointer may be freed, right?

Reply:
You are right, inserting into `unordered_flat_set` will change the pointer of `UStringVal` but the pointer of data won't be changed. We should remove this part of code and it seems not efficient for memory in less than or equal 8 bytes scenario.